File: 283_move_zeros/solution_2/move_zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int last_non_zero = -1;
        for(int i = 0; i < nums.size(); i++) {
            if(nums[i] == 0)
                continue;
            swap(nums[last_non_zero+1], nums[i]);
            last_non_zero++;
        }
    }
};

File: 283_move_zeros/solution_2/move_zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        last_non_zero = -1
        for i in range(len(nums)):
            if nums[i] == 0:
                continue

            t1 = nums[last_non_zero+1]
            nums[last_non_zero+1] = nums[i]
            nums[i] = t1
            last_non_zero += 1

